zk-sdk/src/sigma_proofs/mod.rs

```diff
@@ -66,10 +66,19 @@ use {
 fn ristretto_point_from_optional_slice(
     optional_slice: Option<&[u8]>,
 ) -> Result<CompressedRistretto, SigmaProofVerificationError> {
-    optional_slice
-        .and_then(|slice| (slice.len() == RISTRETTO_POINT_LEN).then_some(slice))
-        .map(CompressedRistretto::from_slice)
-        .ok_or(SigmaProofVerificationError::Deserialization)
+    let Some(slice) = optional_slice else {
+        return Err(SigmaProofVerificationError::Deserialization);
+    };
+
+    if slice.len() != RISTRETTO_POINT_LEN {
+        return Err(SigmaProofVerificationError::Deserialization);
+    }
+
+    let Ok(compressed_ristretto) = CompressedRistretto::from_slice(slice) else {
+        return Err(SigmaProofVerificationError::Deserialization);
+    };
+
+    Ok(compressed_ristretto)
```

Comment:
An explicit `len()` check, I think, is improving readability.
But the last step can be shorter and still be readable:

```suggestion
    let Some(slice) = optional_slice else {
        return Err(SigmaProofVerificationError::Deserialization);
    };

    if slice.len() != RISTRETTO_POINT_LEN {
        return Err(SigmaProofVerificationError::Deserialization);
    }

    CompressedRistretto::from_slice(slice)
        .map_err(|_| SigmaProofVerificationError::Deserialization))
```

---

Note that it is considered suboptimal to combine changes that are unrelated in a single commit.

This PR is already pretty big, and it is about adjusting the code base due to the API changes.  Not about refactoring to make this code more readable.

While it is reasonable to fix locations that become more complex due to the API updates, we should probably keep other changes to a minimum.
And if there is a desire to refactor something, it would be much better to do it in a separate change.